doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+these datastructures available to routers in the same AS, clients in the AS, as well as clients from a remote AS.
+Information for FABRID is also included in the PCBs emitted by each FABRID enabled AS.
+
+The control service introduces a FABRID service with the following endpoints:
+
+- GetMPLSMapIfNecessary (intra-AS)
+- GetRemotePolicyDescription (intra-AS)
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The FABRID service uses the following important data structures:
+
+- SupportedIndicesMap
+    Maps a connection pair consisting of two ConnectionPoints (Type: string, IP: string, Prefix: uint32, InterfaceId: uint16) to a list of policy indices. This map shows for each connection pair which policy indices are supported, which can be one or multiple policies.
+- IndexIdentifierMap
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size. The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers). The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces, a mapping is required from policy index to local and global identifiers. This mapping is provided by this map.
+- IdentifierDescriptionMap
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS. This map maps a local policy identifier to its corresponding description.
+- MPLSMap
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to enforce the policy in the internal network. The MPLSMap data structure stores a mapping between policy indices and the integer MPLS label that is applied to the packet at the border router. Routers periodically fetch this map from the control service. A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- RemotePolicyCache
+    When a local policy is queried at a remote AS, the resulting policy description is cached at the requesting AS’ FABRID Manager, such that subsequent requests can be served from cache.
+
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The IndexIdentifierMap and SupportedIndicesMap are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
```

Comment:
Not related to the document, but I still wonder if we could do anything better than sorting. This seems interesting, although a bit more complex to implement; https://crypto.stackexchange.com/a/54546